crates/oxc_linter/src/rules/unicorn/prefer_string_raw.rs

```diff
@@ -0,0 +1,282 @@
+            return;
+        };
+
+        let parent_node = ctx.nodes().parent_node(node.id());
+
+        if let Some(parent_node) = parent_node {
+            match parent_node.kind() {
+                AstKind::Directive(_) => {
+                    return;
+                }
+                AstKind::ImportDeclaration(decl) => {
+                    if string_literal.span == decl.source.span {
+                        return;
+                    }
+                }
+                AstKind::ExportNamedDeclaration(decl) => {
+                    if let Some(source) = &decl.source {
+                        if string_literal.span == source.span {
+                            return;
+                        }
+                    }
+                }
+                AstKind::ExportAllDeclaration(decl) => {
+                    if string_literal.span == decl.source.span {
+                        return;
+                    }
+                }
+                AstKind::ObjectProperty(prop) => {
+                    let PropertyKey::StringLiteral(key) = &prop.key else {
+                        return;
+                    };
+
+                    if !prop.computed && string_literal.span == key.span {
+                        return;
+                    }
+                }
+                AstKind::PropertyKey(_) => {
+                    if let Some(AstKind::ObjectProperty(prop)) =
+                        ctx.nodes().parent_node(parent_node.id()).map(AstNode::kind)
+                    {
+                        let PropertyKey::StringLiteral(key) = &prop.key else {
+                            return;
+                        };
+
+                        if !prop.computed && key.span == string_literal.span {
+                            return;
+                        }
+                    }
+                }
+                AstKind::JSXAttributeItem(attr) => {
+                    let Some(attr) = attr.as_attribute() else {
+                        return;
+                    };
+
+                    let Some(JSXAttributeValue::StringLiteral(value)) = &attr.value else {
+                        return;
+                    };
+
+                    if value.span == string_literal.span {
+                        return;
+                    }
+                }
+                AstKind::TSEnumMember(member) => {
+                    if member.span == string_literal.span {
+                        return;
+                    };
+
+                    let TSEnumMemberName::String(id) = &member.id else {
+                        return;
+                    };
+
+                    if id.span == string_literal.span {
+                        return;
+                    }
+                }
+                _ => {}
+            }
+        }
+
+        let raw = ctx.source_range(string_literal.span);
+
+        let last_char_index = raw.len() - 2;
+        if let Some('\\') = raw.char_at(Some(last_char_index as f64)) {
```

Comment:
mind explaining to me what this line does? i can't work it out

like i know its checking to see if the last char in the string is a `\` but why?

Reply:
ah i see this in the original impl, any ideas why?

Reply:
Apparently you can still escape backticks in String.raw so if the raw value end's in `\` it will result in an unterminated string.

<img width="976" alt="image" src="https://github.com/user-attachments/assets/8cf3f840-9e15-43c4-9f0e-cb25b1f0a7b9">